lib/protocol/src/openctx-protocol.schema.ts

```diff
@@ -62,11 +69,32 @@ export interface Selector {
      */
     contentContains?: string
 }
-export interface ItemsParams {
+/**
+ * The viewer context sent by client to the provider. 
+ */
+export interface Viewer {
+    /**
+     * The list of respositoy remotes from the workspace.
+     * Eg. ["github.com/sourcegraph/openctx"]
+     */
+    repositoryRemotes?: string[]    
+}
+export interface CandidateItemsParams<K extends ItemKindID = ItemKindID> {
+    /**
+     * A search query that is interpreted by providers to find the candidates items.
+     */
+    query?: string
+    kind?: K
+    viewer?: Viewer
```

Comment:
when we chatted you prefered adding this as an argument to the function. Did it end up cleaner to just use the Params field? TBH params seems like a reasonable place given the http layer.

Reply:
I think we should just keep this `{[key:string]: string}` for now and use a few well-known keys we agree on. Too soon to promote things into the protocol level yet before we've made a few more. For example I am not confident that repositoryRemotes or codebaseNames is the right abstraction (as mentioned above).